src/services/app/user.ts

```diff
@@ -404,21 +406,22 @@ export default class UserService {
     }
 
     public static async existsAccountByPhone(phone: string): Promise<boolean> {
-        const exists = await this.appUserTrust.findOne({
-            attributes: ['phone'],
-            where: { phone },
-            include: [
-                {
-                    model: this.user,
-                    as: 'throughTrust',
-                    where: {
-                        active: true,
-                    },
-                },
-            ],
-            raw: true,
+        const query = `
+            SELECT phone, address
+            FROM app_user_trust
+            LEFT JOIN app_user_through_trust ON "appUserTrustId" = id
+            LEFT JOIN "user" ON "user".address = "userAddress"
+            WHERE phone = :phone
+            AND "user".active = TRUE`;
```

Comment:
hmm, interesting. Was the previous query that big?

Reply:
yes.. this is the query running in production for all requests to /auth:
`SELECT "AppUserTrustModel".*,
         "throughTrust"."address" AS "throughTrust.address",
         "throughTrust"."avatarMediaId" AS "throughTrust.avatarMediaId",
         "throughTrust"."username" AS "throughTrust.username",
         "throughTrust"."language" AS "throughTrust.language",
         "throughTrust"."currency" AS "throughTrust.currency",
         "throughTrust"."pushNotificationToken" AS "throughTrust.pushNotificationToken",
         "throughTrust"."gender" AS "throughTrust.gender",
         "throughTrust"."year" AS "throughTrust.year",
         "throughTrust"."children" AS "throughTrust.children",
         "throughTrust"."lastLogin" AS "throughTrust.lastLogin",
         "throughTrust"."suspect" AS "throughTrust.suspect",
         "throughTrust"."active" AS "throughTrust.active",
         "throughTrust"."createdAt" AS "throughTrust.createdAt",
         "throughTrust"."updatedAt" AS "throughTrust.updatedAt",
         "throughTrust->AppUserThroughTrustModel"."userAddress" AS "throughTrust.AppUserThroughTrustModel.userAddress",
         "throughTrust->AppUserThroughTrustModel"."appUserTrustId" AS "throughTrust.AppUserThroughTrustModel.appUserTrustId"
FROM 
    (SELECT "AppUserTrustModel"."phone",
         "AppUserTrustModel"."id" AS "id"
    FROM "app_user_trust" AS "AppUserTrustModel"
    WHERE "AppUserTrustModel"."phone" = $1
            AND 
        (SELECT "AppUserThroughTrustModel"."userAddress"
        FROM "app_user_through_trust" AS "AppUserThroughTrustModel"
        INNER JOIN "user" AS "UserModel"
            ON "AppUserThroughTrustModel"."userAddress" = "UserModel"."address"
                AND "UserModel"."active" = $2
        WHERE ("AppUserTrustModel"."id" = "AppUserThroughTrustModel"."appUserTrustId") LIMIT $3 ) IS NOT NULL LIMIT $4) AS "AppUserTrustModel"
    INNER JOIN ( "app_user_through_trust" AS "throughTrust->AppUserThroughTrustModel"
    INNER JOIN "user" AS "throughTrust"
        ON "throughTrust"."address" = "throughTrust->AppUserThroughTrustModel"."userAddress")
    ON "AppUserTrustModel"."id" = "throughTrust->AppUserThroughTrustModel"."appUserTrustId"
        AND "throughTrust"."active" = $5`

Reply:
ahh I see the sub-selects. Crap. Had many problems with that in the past. Sequelize is so tricky on those things.